dask/dataframe/tests/test_groupby.py

```diff
@@ -3190,3 +3190,23 @@ def test_groupby_slice_getitem(by, slice_key):
     # column projection after read_parquet etc
     assert hlg_layer(got.dask, "getitem")
     assert_eq(expect, got)
+
+
+def test_groupby_tuple_key():
+    df = pd.DataFrame(
+        {
+            ("a", "b"): [1, 1, 2, 2],
+            "a": [1, 1, 1, 2],
+            "b": [1, 2, 2, 2],
+            "c": [1, 1, 1, 1],
+        }
+    )
+    ddf = dd.from_pandas(df, npartitions=2)
+
+    expected = df.groupby(("a", "b")).c.count()
+
+    with warnings.catch_warnings():
+        if not PANDAS_GT_150:
+            warnings.simplefilter("ignore", category=np.VisibleDeprecationWarning)
```

Comment:
I'd prefer to move this inside our groupby code so users don't get this warning

Reply:
That makes sense to me - just noting that users will still see this warning at compute time once the underlying pandas APIs are called, so we'll still need to filter warnings on the `assert_eq` call here